### PR TITLE
Aoaf 3 login and authentication to the zip portal

### PR DESCRIPTION
**Pull request by okoye-companieshouse**

### CODE REVIEW

File: src/middleware/company.authentication.middleware.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { NextFunction, Request, Response } from "express";
+import { authMiddleware, AuthOptions } from "@companieshouse/web-security-node";
+import { CHS_URL } from "../utils/properties";
+import { urlParams } from "../types/page.urls";
+
+export const companyAuthenticationMiddleware = (req: Request, res: Response, next: NextFunction) => {
+
+    const companyNumber: string = req.params[urlParams.PARAM_ZIP_NUMBER];
```
Comment (amartin7663) [inline src/middleware/company.authentication.middleware.ts:8]: Shouldn't this be PARAM_COMPANY_NUMBER

File: src/types/page.urls.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+export enum urlParams {
+    PARAM_ZIP_NUMBER = "zipNumber",
+    PARAM_COMPANY_NUMBER = "companyNumber",
+    PARAM_TRANSACTION_ID = "transactionId",
+    PARAM_SUBMISSION_ID = "submissionId",
+}
+
+export enum URL_QUERY_PARAM {
```
Comment (amartin7663) [inline src/types/page.urls.ts:8]: Every value in this enum is also in the above one, is there a reason we need both?

File: src/utils/url.ts [hunk 3]
```diff
@@ -0,0 +1,112 @@
+import { Request } from "express";
```
Comment (amartin7663) [inline src/utils/url.ts:1]: Do we use anything in this file anywhere? If not why do we need it?

File: views/includes/signout-bar.html [hunk 4]
```diff
@@ -0,0 +1,11 @@
+{% if userEmail %}
+    {% set signoutURL = signoutURL | default("/appoint-update-remove-company-officer/signout") %}
```
Comment (amartin7663) [inline views/includes/signout-bar.html:2]: That default signout url is wrong

File: views/layout.html [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+{% endblock %}
+
+{% block bodyStart %}
+  {% include "includes/cookie-consent-banner.html" %}
+{% endblock %}
+
+{% block header %}
+  {{ govukHeader({
+    homepageUrl: "https://www.gov.uk",
+    serviceName: "Remove a company director",
```
Comment (amartin7663) [inline views/layout.html:51]: Service name and url is incorrect

File: views/start.html [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+{% block content %}
+  <div class="govuk-grid-row">
+    <div class="govuk-grid-column-two-thirds">
+
+      <h1 class="govuk-heading-xl">
+        Remove a company director
+      </h1>
+
+      {% if FEATURE_FLAG_REMOVE_DIRECTOR_20022023 === "true" %}
+        <p class="govuk-body">Use this service to remove (terminate the appointment of) a director of a limited company on the Companies House register, including corporate directors.</p>
```
Comment (amartin7663) [inline views/start.html:29]: I guess this is placeholder test for Harry's story? We could clean it up though, remove the feature flag etc...

File: src/controllers/error.controller.ts [hunk 7]
```diff
@@ -0,0 +1,17 @@
+import { NextFunction, Request, Response } from "express";
```
Comment (amartin7663) [inline src/controllers/error.controller.ts:1]: There is no unit test for this controller

File: src/middleware/authentication.middleware.ts [hunk 8]
```diff
@@ -0,0 +1,14 @@
+import { NextFunction, Request, Response } from "express";
```
Comment (amartin7663) [inline src/middleware/authentication.middleware.ts:1]: Unit test?

File: src/middleware/company.authentication.middleware.ts [hunk 9]
```diff
@@ -0,0 +1,17 @@
+import { NextFunction, Request, Response } from "express";
+import { authMiddleware, AuthOptions } from "@companieshouse/web-security-node";
+import { CHS_URL } from "../utils/properties";
+import { urlParams } from "../types/page.urls";
+
+export const companyAuthenticationMiddleware = (req: Request, res: Response, next: NextFunction) => {
```
Comment (amartin7663) [inline src/middleware/company.authentication.middleware.ts:6]: Unit test?

File: src/utils/properties.ts [hunk 10]
```diff
@@ -0,0 +1,36 @@
+/**
+ * Gets an environment variable. If the env var is not set and a default value is not
+ * provided, then it is assumed it is a mandatory requirement and an error will be
+ * thrown.
+ */
+
+const getEnvironmentVariable = (key: string, defaultValue?: unknown): string => {
```
Comment (amartin7663) [inline src/utils/properties.ts:7]: There's a few if checks in here so could do a unit test

File: src/utils/url.ts (hunk 3 above)
Comment (okoye-companieshouse) [reply src/utils/url.ts:1]: This is not used at the moment for this WI, will take it out for now.

**Review by amartin7663 — APPROVED** [review]

**Pull request merged**